### PR TITLE
Replace deprecated String.prototype.substr()

### PR DESCRIPTION
**Pull request by CommanderRoot**
[String.prototype.substr()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) is deprecated so we replace it with [String.prototype.slice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) which works similarily but isn't deprecated.
.substr() probably isn't going away anytime soon but the change is trivial so it doesn't hurt to do it.

### CODE REVIEW

**Review by tcitworld — APPROVED** [review]
Thanks for making IDEs happy! 🙂

**Review by raimund-schluessler — APPROVED** [review]

File: src/store/calendars.js [hunk 1]
```diff
@@ -1031,8 +1031,8 @@ const actions = {
 		// calling this action to properly handle it
 		const objectPath = atob(objectId)
 		const lastSlashIndex = objectPath.lastIndexOf('/')
-		const calendarPath = objectPath.substr(0, lastSlashIndex + 1)
-		const objectFileName = objectPath.substr(lastSlashIndex + 1)
+		const calendarPath = objectPath.slice(0, lastSlashIndex + 1)
+		const objectFileName = objectPath.slice(lastSlashIndex + 1)
```
Comment (st3iny) [inline src/store/calendars.js:1035]: Needs to be adjusted.

File: src/utils/color.js [hunk 2]
```diff
@@ -98,9 +98,9 @@ export function detectColor(color) {
 	} else if (/^((?:[A-Fa-f0-9]{3}){1,2})$/.test(color)) { // ff00ff and f0f
 		return '#' + color
 	} else if (/^(#)((?:[A-Fa-f0-9]{8}))$/.test(color)) { // #ff00ffff and #f0ff
-		return color.substr(0, 7)
+		return color.slice(0, 7)
 	} else if (/^((?:[A-Fa-f0-9]{8}))$/.test(color)) { // ff00ffff and f0ff
-		return '#' + color.substr(0, 6)
+		return '#' + color.slice(0, 6)
```
Comment (st3iny) [inline src/utils/color.js:103]: Needs to be adjusted.

**Review by st3iny — CHANGES_REQUESTED** [review]
**Warning:** `substr()` and `slice()` don't behave in the same way!

```
substr(start)
substr(start, length)
```

```
slice(indexStart)
slice(indexStart, indexEnd)
```

The method `substr()` takes a length as its second param while `slice()` takes an index. Usages with only the first argument behave the same way and don't need to be refactored.

**Review by st3iny — APPROVED** [review]
Nevermind, all extended usages start with a 0 index and thus behave in the same way.

**Pull request merged**